### PR TITLE
Added mlops-stepfunctions module

### PR DESCRIPTION
**Pull request by mahesh-balumuri**
## Describe your changes
Added Module which demonstrates implementation of MLOps using AWS Step Functions

## Checklist before requesting a review

- [ x] I updated `CHANGELOG.MD` with a description of my changes
- [ x] If the change was to a module, I ran the code validation script (`scripts/validate.sh`)
- [ x] If the change was to a module, I have added thorough tests
- [ x] If the change was to a module, I have added/updated the module's README.md
- [ x] If a module was added, I added a reference to the module to the repository's README.md
- [ x] I verified that my code deploys successfully using `seedfarmer apply`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 6cc6fa2d17ebdf66b294b0be55c406f3e50cc034
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=299bfa08-9c25-4297-9fbe-f726fc363451%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 6cfb268552b7dacf3182d6b4a1cd762e8d1a6fe4
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c7800545-1c68-44e9-a0ba-24b00acd3d6f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: f22f53b95146e13e77bc4bf73daf3ea1965accd1
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=6dbc33b5-46d7-4cd3-9daf-e74115a53fc1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 71c6585f1661e453700de98cab94819e6b94a6f1
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=6f78f1bf-a8a1-4838-a44d-4e1b723c5522%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/examples/mlops-stepfunctions/app.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from sagemaker.sklearn import defaults  # type: ignore[import-untyped]
+
+from settings import ApplicationSettings
+from stack import MLOPSSFNResources
+
+app = App()
+
+try:
+    app_settings = ApplicationSettings()
+except ValidationError as e:
```
Comment (kukushking) [inline modules/examples/mlops-stepfunctions/app.py:18]: Nit - this try/except doesn't seem to be doing anything. Shall we remove it?

File: modules/examples/mlops-stepfunctions/scripts/input.yaml [hunk 2]
```diff
@@ -0,0 +1,187 @@
+app_id: aiops
```
Comment (kukushking) [inline modules/examples/mlops-stepfunctions/scripts/input.yaml:1]: Does this make sense to mount this config via [dataFiles](https://seed-farmer.readthedocs.io/en/latest/manifests.html#a-word-about-datafiles)?

File: modules/examples/mlops-stepfunctions/app.py (hunk 1 above)
Comment (mahesh-balumuri) [reply modules/examples/mlops-stepfunctions/app.py]: Removed.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: c23dbd344f5ee6d5b089983fa39c4377ec63ab5f
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cf29f917-e6a6-4adf-bd4b-dd3de70c8f1f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 8bee1572d24440e98376e05cc60f279163516ae4
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=246b6bd9-d7d1-4607-813a-82c12e87a170%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 9addfc7b1a174b65035328fc34a1795e51b38a35
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ba9d8926-391e-43e0-a326-53dede8f0f41%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/examples/mlops-stepfunctions/scripts/input.yaml (hunk 2 above)
Comment (mahesh-balumuri) [reply modules/examples/mlops-stepfunctions/scripts/input.yaml]: Updated with the dataFiles.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 1d69f0a7e1ae5fcf8bb836cb91ce1c623b56c107
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e25e0a07-1111-4278-b500-02a933d58959%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: acaf97bae3c7f9a337e0551a441ec43055d291d3
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=20f5a77d-fd4f-49f1-b045-73062ce249dc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/examples/mlops-stepfunctions/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+## Introduction
```
Comment (kukushking) [inline modules/examples/mlops-stepfunctions/README.md:1]: Can you please document module input/output parameters, manifest example, and metadata, following the convention established by all other modules?

File: modules/examples/mlops-stepfunctions/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Assuming that you will be using a single account, follow the guide [here](https://seed-farmer.readthedocs.io/en/latest/bootstrapping.html#) to bootstrap your account(s) to function as a toolchain and target account.
+
+Following is the command to bootstrap your existing account to a toolchain and target account.
+```
+seedfarmer bootstrap toolchain --project aiops --trusted-principal ${ADMIN_ROLE_ARN} --as-target
+```
+
+## Deployment
+
+Pick the manifest to deploy. Manifests are located in `manifests/` directory. For example, to deploy this modules, run:
```
Comment (kukushking) [inline modules/examples/mlops-stepfunctions/README.md:73]: Deployment guide is common for all modules, we do not have to repeat it in the individual module readme. You can add a reference to the generic deployment guide.

File: modules/examples/mlops-stepfunctions/scripts/input.yaml (hunk 2 above)
Comment (kukushking) [reply modules/examples/mlops-stepfunctions/scripts/input.yaml]: Reverting back on this. Given the training scrips are part of the package, we gain very limited flexibility in using data files for the step function. I think it makes sense to keep it inside the module. Please revert.

File: modules/examples/mlops-stepfunctions/stack.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        lambda_role.attach_inline_policy(aws_iam.Policy(self, "S3AccessRole", document=s3_access_statements))
+
+        # Create the EventBridge rule
+
+        event_rule = events.Rule(
+            self,
+            "MyEventRule",
+            schedule=events.Schedule.cron(
+                minute="0",
+                hour="18",  # 6 PM UTC
```
Comment (kukushking) [inline modules/examples/mlops-stepfunctions/stack.py:178]: Should we parametrize this?

File: modules/examples/mlops-stepfunctions/README.md (hunk 4 above)
Comment (mahesh-balumuri) [reply modules/examples/mlops-stepfunctions/README.md]: Updated.

File: modules/examples/mlops-stepfunctions/README.md (hunk 3 above)
Comment (mahesh-balumuri) [reply modules/examples/mlops-stepfunctions/README.md:1]: Updated.

File: modules/examples/mlops-stepfunctions/stack.py (hunk 5 above)
Comment (mahesh-balumuri) [reply modules/examples/mlops-stepfunctions/stack.py]: Done.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 6f79563d0378d0348ddf9be3c057764e21533ae3
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=09e710bd-4f0b-4b32-8b75-5f4485e5f3b6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: f89146e3b9ff8737ab278b517a25086c3f913996
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e6ab7c0b-18b7-4034-bb66-48bfcd65b654%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 0245f95493fc93885aaf3cdc40004801cea412cb
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=2decfdb7-314e-41c1-b4ed-d68e196602e1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: dce29ffa3ce4583e46913360875b3e1695a5068b
* Result: FAILED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=144d3798-56ab-4150-8f38-8962a8df176f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: dce29ffa3ce4583e46913360875b3e1695a5068b
* Result: FAILED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=144d3798-56ab-4150-8f38-8962a8df176f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/examples/mlops-stepfunctions/scripts/input.yaml (hunk 2 above)
Comment (mahesh-balumuri) [reply modules/examples/mlops-stepfunctions/scripts/input.yaml:1]: Reverted.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: b070f713eb24aa5029f107a88708d1049306b507
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c32c38c2-a3b6-4acb-b031-b4764d0b3048%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 25ad85fa61dd1376a53864440c1bf66fe623d137
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=80a58282-fc99-40ac-a12e-7d5671234f17%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/examples/mlops-stepfunctions/setup.cfg [hunk 6]
```diff
@@ -0,0 +1,29 @@
+[metadata]
```
Comment (kukushking) [inline modules/examples/mlops-stepfunctions/setup.cfg:1]: Please remove this file as it's duplicating pyproject.toml

File: modules/examples/mlops-stepfunctions/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,211 @@
+#
+# This file is autogenerated by pip-compile with Python 3.9
+# by the following command:
+#
+#    pip-compile --output-file=requirements.txt requirements.in
```
Comment (kukushking) [inline modules/examples/mlops-stepfunctions/requirements.txt:5]: Can you please check-in requirements.in?

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 5802d7d11f1558b53cda4419bfc64d153767dac3
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=df51e4c9-a7e1-4a1d-9fff-7dd5705fdb72%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 85c1acce538e229017c0554024163ce7fc231c8c
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c6f2605e-51c5-4fb8-bb47-6b0439a3f7dc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by LeonLuttenberger — APPROVED** [review]

**Review by kukushking — APPROVED** [review]

**Pull request merged**